bpf/bpf_network.c

```diff
@@ -57,13 +68,21 @@ int from_network(struct __ctx_buff *ctx)
  */
 #ifdef ENABLE_IPSEC
 	if ((ctx->mark & MARK_MAGIC_HOST_MASK) == MARK_MAGIC_DECRYPT)
-		send_trace_notify(ctx, TRACE_FROM_NETWORK, get_identity(ctx), 0, 0,
```

Comment:
I removed this `get_identity` because IIUC there is no possible case that the mark contains the identity in this context.

Reply:
@YutaroHayakawa I'm a bit nervous about this. If we know that `bpf_network.o` is only ever attached directly to interfaces coming in from the outside world, and there's no way to loop traffic back into that interface from inside the node (eg after encryption), then I agree. However, the code says that the mark includes `MARK_MAGIC_DECRYPT`. So that says that Cilium has controlled the previous path, set the `MARK_MAGIC_DECRYPT`, and hence we trigger this particular codepath. So, that implies that maybe Cilium did actually also leave the identity in the mark?

One practical way to confirm or deny this would be to set up an encryption setup, (not sure whether it needs to be tunnel or not), and watch these trace_notify messages to see whether the identity of the source shows up or not.